quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/PropertyMappingInterceptor.java

```diff
@@ -123,11 +135,11 @@ public Iterator<String> iterateNames(ConfigSourceInterceptorContext context) {
 
         // include anything remaining that has a default value
         var defaultStream = allMappers.stream()
-                .filter(m -> !m.getDefaultValue().isEmpty() && !m.hasWildcard()
+                .filter(m -> m.getDefaultValue().isPresent() && !m.hasWildcard()
                         && m.getCategory() != OptionCategory.CONFIG) // advertising the keystore type causes the keystore to be used early
                 .flatMap(m -> toDistinctStream(m.getTo()));
 
-        return IteratorUtils.chainedIterator(baseStream.iterator(), defaultStream.iterator());
+        return IteratorUtils.chainedIterator(baseStream.distinct().iterator(), defaultStream.iterator());
```

Comment:
It should be possible to use allMappers to do this inline if we want to avoid this extra iteration.

Reply:
@shawkins Do you have anything in mind on how can I do this? What extra iteration? Thanks